app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -102,6 +103,19 @@ class Bolt_Boltpay_Model_Payment extends Mage_Payment_Model_Method_Abstract
         self::HOOK_TYPE_VOID => self::TRANSACTION_CANCELLED
     );
 
+
+    /**
+     * Bolt_Boltpay_Model_Payment constructor.
+     *
+     * Allows transitions from on-hold from the non-webhook context
+     */
+    public function __construct()
+    {
+        if (!Bolt_Boltpay_Helper_Api::$fromHooks) {
+            $this->_validStateTransitions[self::TRANSACTION_ON_HOLD] = array(self::TRANSACTION_ALL_STATES);
```

Comment:
instead of introducing new magic state, can we do:
`array_push($this->_validStateTransitions[self::TRANSACTION_ON_HOLD], self::TRANSACTION_COMPLETED, self::TRANSACTION_PENDING, self::TRANSACTION_AUTHORIZED))`

Reply:
@daisy1754 That was my first thought.  I believe the magic state is better because clients may have several states that they wish to transition manually which may lie outside any bounds which we may define.

In anticipation of this, the second magic state was introduced, like the TRANSACTION::NO_NEW_STATE

Reply:
I agree with @mccottry on magic states.

Reply:
Hmm so it sounds like we should allow transition from *any* state to *any* state for manual update, shouldn't we?

Reply:
@daisy1754 I believe that is true and constitutes a change request that is created in a different branch.  It is not related to unholding.